mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -219,6 +219,8 @@
 
     /**
      * Searches for mailboxes matching the given query.
+     *
+     * Caller is responsible to close this stream to ensure resources and search context get released.
```

Comment:
Why this warning here and not on every other methods?

Reply:
not fixed?

Reply:
See https://github.com/linagora/james-project/pull/2919#issuecomment-558482715

Reply:
https://github.com/linagora/james-project/pull/2919#issuecomment-558549766

> > However, if we are confident about resource leaks, we can adopt that PR

https://github.com/linagora/james-project/pull/2919#issuecomment-558552465

> From my knowledge ES is the only place we rely on that stream feature.... Other places are streams from in memory data sources.

And the minimum we can do is document this specific behaviour ;-)